hack/test-cmd.sh

```diff
@@ -74,6 +74,25 @@ function check-curl-proxy-code()
   echo "For address ${full_address}, got ${status} but wanted ${desired}"
   return 1
 }
+
+# TODO: Remove this function when we do the retry inside the kubectl commands. See #15333.
+function kubectl-with-retry()
+{
+  retry=true
+  count=0
+  ERROR_FILE="${KUBE_TEMP}/kubectl-error"
+  while $retry && [[ $count -lt 4 ]]; do
```

Comment:
how about `for count in $(seq 1 4); do` and then `break` when you don't need to retry?